src/utils/daos_hdlr.c

```diff
@@ -1645,6 +1650,73 @@ fs_copy_file(struct cmd_args_s *ap,
 	return rc;
 }
 
+static int
+fs_copy_symlink(struct cmd_args_s *ap,
+		struct file_dfs *src_file_dfs,
+		struct file_dfs *dst_file_dfs,
+		struct stat *src_stat,
+		const char *src_path,
+		const char *dst_path)
+{
+	int		rc = 0;
+	daos_size_t	len = 0;
+	char		symlink_value[DFS_MAX_PATH];
+
+	len = PATH_MAX - 1;
+	if (src_file_dfs->type == POSIX) {
+		len = readlink(src_path, symlink_value, (size_t)len);
+		if (len < 0) {
+			DH_PERROR_DER(ap, rc, "fs_copy_symlink failed on readlink('%s')",
+				      src_path);
+			D_GOTO(out_copy_symlink, rc = daos_errno2der(errno));
```

Comment:
You need to log the same error as you return, also very few function are guaranteed to leave errno untouched.
```suggestion
			int err = errno;

			DH_PERROR_DER(ap, err, "fs_copy_symlink failed on readlink('%s')", src_path);
			D_GOTO(out_copy_symlink, rc = daos_errno2der(err));
```

Reply:
fixed. thx!